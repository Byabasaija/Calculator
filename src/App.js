/* eslint-disable react/no-unused-prop-types */
import './App.css';

function App() {
  return (
    <div className="App">
      <h1> Calculator</h1>
      <p>Welcome to my calculator</p>
    </div>
  );
}
export default App;
