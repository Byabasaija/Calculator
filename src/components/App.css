.App {
    text-align: center;
    width: 700px;
    margin-left: 300px;
    margin-top: 20px;
    background: #f1f2f6;
    padding: 20px 10px;
    border-radius: 10px;
  }
  
  .flex {
    display: flex;
  }
  