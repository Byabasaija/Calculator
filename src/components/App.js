import React from 'react';
import Display from './Display';
import ButtonPannel from './ButtonPannel';
import calculate from '../logic/calculate';

calculate({ total: '10', next: '5', operation: 'X' }, '=');

const App = () => (
  <>
    <Display />
    <ButtonPannel />
  </>
);

export default App;
