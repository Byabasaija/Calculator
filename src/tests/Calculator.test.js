import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App';

it('renders calculator correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
