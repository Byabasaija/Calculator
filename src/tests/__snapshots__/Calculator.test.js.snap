// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator correctly 1`] = `
<div
  className="flex"
>
  <h2>
    Let's do some math
  </h2>
  <div
    className="App"
  >
    <div
      className="Display"
    >
      
    </div>
    <div
      className="ButtonPannel"
    >
      <div
        className="btn-group"
      >
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn  btn--orange"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn  btn--orange"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn  btn--orange"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn  btn--orange"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="btn btn--span-2 "
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn  "
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn  btn--orange"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
